proto/spire/api/agent/delegatedidentity/v1/delegatedidentity.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+package spire.api.agent.delegatedidentity.v1;
+option go_package = "github.com/spiffe/spire-api-sdk/proto/spire/api/agent/delegatedidentity/v1;delegatedidentityv1";
+
+import "spire/api/types/selector.proto";
+import "spire/api/types/x509svid.proto";
+
+// The delegatedIdentity service provides an interface to get the SVIDs of other
+// workloads on the host. This service is intended for use cases where a process
+// (different than the workload one) should access the workload's SVID to
+// perform actions on behalf of the workload. One example of is using a single
+// node instance of Envoy that upgrades TCP connections for different processes
+// running in such a node.
+//
+// The caller must be local and its identity must be listed in the allowed
+// clients on the spire-agent configuration.
+service DelegatedIdentity {
+    // Subscribe to get X.509-SVIDs for identities that match the given selectors.
+    // The lifetime of the subscription aligns to the lifetime of the stream.
+    rpc SubscribeToX509SVIDs(SubscribeToX509SVIDsRequest) returns (stream SubscribeToX509SVIDsResponse);
+
+    // Subscribe to get local and federated bundle.
+    // The lifetime of the subscription aligns to the lifetime of the stream.
+    rpc SubscribeToX509Bundle(SubscribeToX509BundleRequest) returns (stream SubscribeToX509BundleResponse);
+}
+
+// X.509 SPIFFE Verifiable Identity Document with the private key.
+message X509SVIDWithKey {
+    // The agent X509-SVID.
+    spire.api.types.X509SVID x509_svid = 1;
+
+    // Key encoding (DER PKCS#8).
+    bytes x509_svid_key = 2;
+
+    // Names of the trust domains that this workload should federates with.
+    repeated string federates_with = 3;
+}
+
+// SubscribeToX509SVIDsRequest is used by clients to subscribe the set of workloads that
+// it receives SVIDs for.
+message SubscribeToX509SVIDsRequest {
+    // Required. Selectors of the workload to add to the watched set.
+    repeated spire.api.types.Selector selectors = 1;
+}
+
+message SubscribeToX509SVIDsResponse {
+    repeated X509SVIDWithKey x509_svids = 1;
+}
+
+message SubscribeToX509BundleRequest {}
+
+// SubscribeToX509BundleResponse represents the bundle for a given trust domain. An
+// empty bundle indicates that it was removed. Use DER-encoded in X.509 certs.
```

Comment:
> An empty bundle indicates that it was removed.

An alternative could be to have one message that sends all the bundles. When any of them change (or are removed), the entire set is retransmitted. This is closer to the behavior of the workload api (and the behavior of `SubscribeToX509SVIDs` I assume?), but also means the message size will be much larger. I'm not sure how easy it would be to start running into limitations there

Reply:
@evan2645 , we discuss it and we think would be better to send all the bundles in one message too. About sending an empty message is it ok?
About the message size, would be it critical for now? what do you think?

Reply:
Re: message size ... the experience here will be, the integration will break entirely for any node that is running one or more federated workloads where the sum of the federated bundles across all workloads on the node is >~4MB. It can be that the addition of a new federated bundle causes us to cross this threshold (and fail suddenly). How likely this is, I'm not sure ...

We already have this problem on workload api, but it is on a per-workload basis. Here, we are talking about per-node. We should probably figure out message overhead and how many bytes per cert to better understand how realistic this worry is.

Reply:
I think I'd be fine with starting with a "send the world" approach to this at first. We can always add a different RPC later if we determine this is too problematic. I think the "send the world" approach is going to be simpler to implement on both sides, and is probably an ok place to start. Certainly we can imagine a world where the number of federation relationships is high enough to trip the message limit, but in most instances it feels contrived.

Reply:
> Certainly we can imagine a world where the number of federation relationships is high enough to trip the message limit, but in most instances it feels contrived.

How many certs do we think here to bust the limit? 100? 1,000? 10,000? The first doesn't seem impossible, the second is unlikely IMO. I think the important thing here is the failure mode and how we handle it. We probably don't want an outage being caused across the fleet because an operator added a new federation relationship.

Reply:
So, a cert over an EC256 public key with a SHA256 hash with minimal extensions is ~300 bytes. A cert over an RSA 2048-bit key of similar complexity is ~700 bytes (somewhat rough ballpark figures). We're probably looking on the order of thousands of certificates before tripping up.

Reply:
👍 thanks @azdagron, sounds like we'll be ok for now.

if we're alright on message size, my vote is for one message with all the bundles.

Reply:
SGTM!